File: engine/transpile/src/test/resources/golden/SYK006/WS_作業項目.py
"""SYK006 のデータ部レコード WS-作業項目 を逐語対訳した自動生成コード。"""
from cobol_runtime import (
    decode_alphanumeric,
    decode_binary,
    decode_packed,
    decode_zoned,
    encode_alphanumeric,
    encode_binary,
    encode_packed,
    encode_zoned,
)


class WS_作業項目:
    """01 WS-作業項目 総バイト長 94。"""

    LENGTH = 94

    def __init__(self, data=None):
        if data is None:
            self._data = bytearray(94)
        else:
            self._data = bytearray(data)
            if len(self._data) < 94:
                self._data.extend(b"\x00" * (94 - len(self._data)))

    @property
    def data(self):
        return self._data

    # WS-処理件数
    def get_WS_処理件数(self):
        return decode_zoned(self._data, 0, 5)
    def set_WS_処理件数(self, value):
        encode_zoned(self._data, 0, 5, False, value)

    # WS-エラー件数
    def get_WS_エラー件数(self):
        return decode_zoned(self._data, 5, 5)
    def set_WS_エラー件数(self, value):
        encode_zoned(self._data, 5, 5, False, value)

    # WS-エラー件数INDEX
    def get_WS_エラー件数INDEX(self):
        return decode_binary(self._data, 10, 2, True)
    def set_WS_エラー件数INDEX(self, value):
        encode_binary(self._data, 10, 2, True, value)

    # WS-メッセージ区分
    def get_WS_メッセージ区分(self):
        return decode_alphanumeric(self._data, 12, 2)
    def set_WS_メッセージ区分(self, value):
        encode_alphanumeric(self._data, 12, 2, value)

    # WS-メッセージ内容
    def get_WS_メッセージ内容(self):
        return decode_alphanumeric(self._data, 14, 80)
    def set_WS_メッセージ内容(self, value):
        encode_alphanumeric(self._data, 14, 80, value)

File: src/engine/transpile/src/test/resources/golden/SYK006/WS_作業項目.java
package cobolinsight.generated;

import cobolinsight.runtime.CobolRuntime;
import java.util.Arrays;

/** SYK006 のデータ部レコード WS-作業項目 を逐語対訳した自動生成コード。 */
public final class WS_作業項目 {

    /** 01 WS-作業項目 総バイト長 94。 */
    public static final int LENGTH = 94;

    private final byte[] data;

    public WS_作業項目() {
        this.data = new byte[94];
    }

    public WS_作業項目(byte[] source) {
        this.data = Arrays.copyOf(source, 94);
    }

    public byte[] data() {
        return data;
    }

    // WS-処理件数
    public long get_WS_処理件数() {
        return CobolRuntime.decodeZoned(data, 0, 5);
    }
    public void set_WS_処理件数(long value) {
        CobolRuntime.encodeZoned(data, 0, 5, false, value);
    }

    // WS-エラー件数
    public long get_WS_エラー件数() {
        return CobolRuntime.decodeZoned(data, 5, 5);
    }
    public void set_WS_エラー件数(long value) {
        CobolRuntime.encodeZoned(data, 5, 5, false, value);
    }

    // WS-エラー件数INDEX
    public long get_WS_エラー件数INDEX() {
        return CobolRuntime.decodeBinary(data, 10, 2, true);
    }
    public void set_WS_エラー件数INDEX(long value) {
        CobolRuntime.encodeBinary(data, 10, 2, true, value);
    }

    // WS-メッセージ区分
    public String get_WS_メッセージ区分() {
        return CobolRuntime.decodeAlphanumeric(data, 12, 2);
    }
    public void set_WS_メッセージ区分(String value) {
        CobolRuntime.encodeAlphanumeric(data, 12, 2, value);
    }

    // WS-メッセージ内容
    public String get_WS_メッセージ内容() {
        return CobolRuntime.decodeAlphanumeric(data, 14, 80);
    }
    public void set_WS_メッセージ内容(String value) {
        CobolRuntime.encodeAlphanumeric(data, 14, 80, value);
    }
}
